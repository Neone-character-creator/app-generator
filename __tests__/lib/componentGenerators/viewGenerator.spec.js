const generateComponent = require("../../../lib/componentGenerators/genericGenerator")();
describe("view component generator", () => {
    const config = {
        name: "Test",
        views: ["summary"],
        components: {
            summary: {
                type: "view",
                children: ["foo"],
            },
            foo: {
                type: "textfield"
            }
        }
    };
    it("renders the child components", () => {
        const generatedComponent = generateComponent("view")("summary")(config);
        expect(generatedComponent).toEqual(expect.stringContaining(`<TextField label="foo" id="foo" value="" />`));
    });
    it("adds imports for each child component", () => {
        const generatedComponent = generateComponent("view")("summary")(config);
        expect(generatedComponent).toEqual(expect.stringContaining(`import FooTextfield from './FooTextfield';`));
    });
    it("throws an error if a child has an invalid type", () => {
        const config = {
            name: "Test",
            views: ["summary"],
            components: {
                foo: {
                    type: "invalid"
                }
            }
        };
        expect(() => {
            generateComponent("view")("summary")(config);
        }).toThrow();
    });
    it("throws an error if a view has no children", () => {
        const config = {
            name: "Test",
            views:
                ["aummary"],
            components: {
                summary: {},
            }
        };
        expect(() => {
            generateComponent("view")("summary")(config);
        }).toThrow();
    });
    it("throws an error for unsupported child type", () => {
        const config = {
            name: "Test",
            views: ["summary"],
            components: {
                summary: {
                    type: "view",
                    children: ["foo", "bar"]
                },
                foo: {
                    type: "some nonsense"
                }
            }
        };
        expect(() => {
            generateComponent("view")("summary")(config)
        }).toThrowErrorMatchingSnapshot();
    });
});