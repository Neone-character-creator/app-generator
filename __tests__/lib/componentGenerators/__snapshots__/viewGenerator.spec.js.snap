// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view component generator throws an error for unsupported child type 1`] = `"No config for summaryview found"`;
